### DISCUSSION TITLE
Differences between Spacy3 and Spacy2 in NER

### DISCUSSION THREAD

**Discussion by ambuje** [explosion/spaCy#12246] [Help: Model Advice] [answered]
Hi, 

I saw a large improvement in the spacy3 models compared to the spacy2 model.

I am using the same default model (eg ja_core_news_lg) and observing that I am getting an accuracy of around 98 in spacy 3 (training using config) and was getting an accuracy of less than 90 in spacy 2 (training using python code)

I have trained a lot of languages and have observed the same that spacy3 is giving f1 about 95. 

Seeing these extraordinary results, I just want to confirm if I am missing something or if anything major has been changed from spacy2 to spacy3.3 in terms of architecture. Or you could help me with the reasons for getting so much better accuracy from spacy3

Thank you,
Ambuje

**Comment by rmitsch** [comment, marked as answer]
Hi @ambuje, glad to hear that v3 improved your results! 

It's difficult to pinpoint the causes for these improvements exactly, as this also depends on your data and your implementation of the training loop in v2. Some relevant changes from v2 to v3 include retrained models, automatic early stopping in v3's training loop implementation used by the `train` CLI, tokenization changes, different defaults etc.
